Makefile

```diff
@@ -118,7 +118,7 @@ vulncheck: ## Verify code vulnerabilities
 	@govulncheck ./pkg/...
 
 licenses: download ## Verifies dependency licenses
-	! go-licenses csv ./... | grep -v -e 'MIT' -e 'Apache-2.0' -e 'BSD-3-Clause' -e 'BSD-2-Clause' -e 'ISC' -e 'MPL-2.0'
+	! go-licenses csv ./... | grep -v -e 'MIT' -e 'Apache-2.0' -e 'BSD-3-Clause' -e 'BSD-2-Clause' -e 'ISC' -e 'MPL-2.0' -e 'github.com/awslabs/amazon-eks-ami/nodeadm'
```

Comment:
So this is weird, this wasn't causing issues and now it suddenly is. The problem is that the go module is not in the repos root with the LICENSE file so it's not detected. This doesn't appear to have ever been supported (https://github.com/google/go-licenses/issues/186) and we were getting errors in CI previously, but strangely the job itself wasn't failing ([ref](https://github.com/aws/karpenter-provider-aws/actions/runs/7909016450/job/21589309233)). There hasn't been any new release of `go-licenses` so this really is a bit of a mystery. I acknowledge this isn't an ideal solution, though we know the license on this package and I think we can be reasonably confident it won't change. An alternative solution may be adding a LICENSE file to the go module root in the nodeadm repo, don't know if there's any precedent for that.

Reply:
So it's not that the license file doesn't exist, it's that it isn't in a known format according to `go-licenses`. If you look here, `go` actual special cases the `LICENSE` file by copying the file (if one exists in the repo root) and one doesn't exist directly next to the `go.mod` file: https://go.dev/ref/mod#vcs-license:~:text=testing%20their%20dependencies.-,Special%20case%20for%20LICENSE%20files,-When%20the%20go. 

I was able to repro this locally by pulling the latest version of `nodeadm` pacakge and see that the LICENSE file is present.

<img width="515" alt="Screenshot 2024-02-21 at 10 32 15 PM" src="https://github.com/aws/karpenter-provider-aws/assets/26334334/f05b9518-0566-411d-aac4-b46bc9904d10">

Reply:
From just messing around with it, it appears that the `eks-ami` repo is not actually using the [official MIT license](https://github.com/git/git-scm.com/blob/main/MIT-LICENSE.txt) which is why `go-licenses` is failing. It's missing this stanza

```
The above copyright notice and this permission notice shall be
included in all copies or substantial portions of the Software.
```

May be worth following up with them to see if they can add the stanza so they conform to the full MIT license.

Reply:
Ah, that makes sense. What's weird to me though is that it was working before, then stopped working. There hasn't been a change to the license in 6 years so there must have been some change in `go-licenses`. Given it started failing after I repulled the toolchain with go 1.22 and deleted the previous toolchain cache I think that's likely the case. I can go ahead and open an issue against the ami repo.

Reply:
We're going to decouple the license fix from this feature, leaving the check in with a TODO to remove once the license has been updated.